jupyterlabutils/notebook/efd_helper.py

```diff
@@ -0,0 +1,142 @@
+# This file is part of jupyterlabutils.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import aioinflux
+from collections import Counter
+import logging
+import pandas as pd
+
+from jupyterlabutils.notebook import NotebookAuth
+
+
+class EfdClient:
+    """Class to handle connections and basic queries"""
+    def __init__(self, efd_name, db_name='efd', port='443', path_to_creds='~/.lsst/notebook_auth.yaml'):
+        self.db_name = db_name
+        self.auth = NotebookAuth(path=path_to_creds)
+        self.host, self.user, self.password = self.auth.get_auth(efd_name)
+        self.client = aioinflux.InfluxDBClient(host=self.host,
+                                               port=port,
+                                               ssl=True,
+                                               username=self.user,
+                                               password=self.password,
+                                               db=db_name,
+                                               mode='async')  # mode='blocking')
+        self.client.output = 'dataframe'
+
+    async def get_topics(self):
+        topics = await self.client.query('SHOW MEASUREMENTS')
+        return topics['name'].tolist()
+
+    async def get_fields(self, topic_name):
+        fields = await self.client.query(f'SHOW FIELD KEYS FROM "{self.db_name}"."autogen"."{topic_name}"')
+        return fields['fieldKey'].tolist()
+
+    async def select_time_series(self, topic_name, fields, t1, t2, is_window=False):
```

Comment:
Would like to be able to handle query predicates too.

Reply:
I've specifically made available the underlying aioinflux client to the user of the class.  I'd prefer to say that if a user wants to execute queries with arbitrary predicates they use that mechanism rather than trying to write a method that could deal with them.

If you think there are specific predicates (other than time window) that will be common, I'd like to hear about them to see if we should wrap those specifically.

Reply:
Fair point. I would recommend splitting the part of `select_time_series` that builds the query into a new API function. That way I can use the existing code to build the query and then add predicates how I want with the query string.

Reply:
This is done by implementing a function called `build_time_range_query`.  The query methods now delegate to that for building the query.  This shows up in  v0.1.6 of teh client on PyPI.